EIPS/eip-6466.md

```diff
@@ -0,0 +1,201 @@
+---
+eip: 6466
+title: SSZ receipts root
+description: Migration of receipts MPT commitment to SSZ
+author: Etan Kissling (@etan-status), Vitalik Buterin (@vbuterin)
+discussions-to: https://ethereum-magicians.org/t/eip-6466-ssz-receipts-root/12884
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-02-08
+requires: 155, 658, 1559, 2718, 2930, 4844, 6404
+---
+
+## Abstract
+
+This EIP defines a migration process of existing Merkle-Patricia Trie (MPT) commitments for receipts to SSZ.
+
+## Motivation
+
+While the consensus `ExecutionPayloadHeader` and the execution block header map to each other conceptually, they are encoded differently. This EIP aims to align the encoding of their fields, taking advantage of the more modern SSZ format. This brings several advantages:
+
+1. **Reducing complexity:** Merkle-Patricia Tries (MPT) are hard to work with. Replacing them with SSZ leaves only the state trie in the legacy MPT format.
+
+2. **Better for smart contracts:** The SSZ format is optimized for production and verification of merkle proofs. It allows proving specific fields of containers and allows chunked processing.
+
+3. **Better for light clients:** Light clients with access to the consensus `ExecutionPayload` no longer need to implement Merkle-Patricia Trie or RLP libraries to work with receipts.
```

Comment:
This looks similar to the Motivation from EIP-6404. Is that intentional?

Reply:
Same motivation for all three: EIP-6404 for transactions, EIP-6465 for receipts, and EIP-6466 for withdrawals. The three MPTs are independent of each other and don't necessarily be linked together.